rust/numaflow-core/src/source/sqs.rs

```diff
@@ -0,0 +1,344 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use chrono::DateTime;
+use numaflow_sqs::source::{
+    sent_timestamp_message_system_attribute_name, SqsMessage, SqsSource, SqsSourceBuilder,
+    SqsSourceConfig,
+};
+
+use crate::config::{get_vertex_name, get_vertex_replica};
+use crate::error::Error;
+use crate::message::{Message, MessageID, Offset, StringOffset};
+use crate::source;
+
+impl TryFrom<SqsMessage> for Message {
+    type Error = Error;
+
+    fn try_from(message: SqsMessage) -> crate::Result<Self> {
+        let offset = Offset::String(StringOffset::new(message.offset, *get_vertex_replica()));
+
+        // find the SentTimestamp header and convert it to a chrono::DateTime<Utc>
+        let sent_timestamp = match message
+            .headers
+            .get(sent_timestamp_message_system_attribute_name())
+        {
+            Some(sent_timestamp_str) => sent_timestamp_str
+                .parse::<i64>()
+                .ok()
+                .and_then(DateTime::from_timestamp_millis),
+            None => None,
+        };
+
+        Ok(Message {
+            keys: Arc::from(vec![message.key]),
+            tags: None,
+            value: message.payload,
+            offset: offset.clone(),
+            event_time: message.event_time,
+            watermark: sent_timestamp,
```

Comment:
What is the difference between sent timestamp and event time? Can you document why we are using sent timestamp for watermark?